No manual unrolling for 256-length loops

kirushik opened the pull request:
As a followup for #1, I decided to bench @Vurich assumption that 256-length unrolling is just too much and will only make things worse by busting the caches.

I've started by adding benchmarks, which showed me the following (please note that the `limit_256` feature flag is still required):
```
test backwards_with_crunchy    ... bench:         284 ns/iter (+/- 23)
test backwards_without_crunchy ... bench:          38 ns/iter (+/- 2)
test forwards_with_crunchy     ... bench:         288 ns/iter (+/- 54)
test forwards_without_crunchy  ... bench:          26 ns/iter (+/- 4)
```

So let's use the usual forward-facing loop here, and rely on compiler to do it's vectorization magic.

kirushik committed:
Pass a feature-flag to increase unrolling limit

kirushik committed:
Benchmarks for loop unrolling

kirushik committed:
Don't unroll 256-step loops, they will be vectorized

debris reviewed (commented):
You are probably right. But can you use `test::black_box` in the benchmarks? I'm just afraid that the compiler is evaluating everything at compile time.

kirushik committed:
Trying to outsmart optimizer in crunchy benches

kirushik commented:
Valid concern.

I've blackboxed the `data` execution result after the loop -- and the numbers remained the same:
```
test backwards_with_crunchy    ... bench:         300 ns/iter (+/- 79)
test backwards_without_crunchy ... bench:          39 ns/iter (+/- 3)
test forwards_with_crunchy     ... bench:         302 ns/iter (+/- 70)
test forwards_without_crunchy  ... bench:          28 ns/iter (+/- 2)
```

If I blackbox every subsequent XOR operation, it ruins the vectorization of the loop, but not using crunchy is still faster:
```
test backwards_with_crunchy    ... bench:         294 ns/iter (+/- 60)
test backwards_without_crunchy ... bench:         218 ns/iter (+/- 8)
test forwards_with_crunchy     ... bench:         298 ns/iter (+/- 24)
test forwards_without_crunchy  ... bench:         226 ns/iter (+/- 20)
```

I've also replaced static vectors of 0s and 1s with randomly generated ones, and moved `other_data` generation insode of the `iter` loop — so optimizer will have no way of doing magic with subsequent XORing the same data. This made results not directly comparable with the ones above, but the difference is still in favor of not using `crunchy`:
```
test backwards_with_crunchy    ... bench:       1,372 ns/iter (+/- 115)
test backwards_without_crunchy ... bench:         858 ns/iter (+/- 94)
test forwards_with_crunchy     ... bench:       1,372 ns/iter (+/- 114)
test forwards_without_crunchy  ... bench:         857 ns/iter (+/- 92)
```
